cpp/unit/unit_texture.cpp

```diff
@@ -145,7 +145,7 @@ unsigned int dir_group(coord::phys3_delta dir, unsigned int angles) {
 	unsigned int first_angle = 5 * angles / 8;
 
 	// normalise direction vector
-	double len = std::hypot(dir.ne, dir.se);
+	double len = std::hypot(dir.ne, dir.se).to_double();
```

Comment:
I was thinking of changing it to this:
```
	coord::phys_t len = std::hypot(dir.ne, dir.se);
	double dir_ne = static_cast<double>(dir.ne / len);
	double dir_se = static_cast<double>(dir.se / len);
```

Or maybe use the to_double function instead of the static_cast:
`double dir_ne = (dir.ne / len).to_double();`

I think having the operators return FixedPoint is nicer, because it's nice to be able to divide by two or add 0.5f, we use that a lot.